src/dotnet/Program.cs

```diff
@@ -138,6 +141,16 @@ internal static int ProcessArgs(string[] args, ITelemetry telemetryClient)
 
         }
 
+        /// <summary>
+        /// Performs initialization that must be done before anything else in the process.
+        /// </summary>
+        private static void InitializeProcess()
+        {
+            // by default, .NET Core doesn't have all code pages needed for Console apps.
+            // see the .NET Core Notes in https://msdn.microsoft.com/en-us/library/system.diagnostics.process(v=vs.110).aspx
+            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
```

Comment:
Reading through some of the other issues I see it appears to be the default Console encoding that is causing the dependency. That is kind of unfortunate that the Console doesn't use a standard algorithm encoding by default.